### PR TITLE
Changes from this sprint

### PR DESCRIPTION
**Pull request by carrollh**

### CODE REVIEW

File: scripts/report_event.py [hunk 1]
```diff
@@ -23,11 +23,13 @@
...
 from SignaliQ.model.CloudProviderEvent import CloudProviderEvent
 from SignaliQ.model.ProviderEventsUpdateMessage import ProviderEventsUpdateMessage
+from SignaliQ.model.CloudVM import CloudVM
+from SignaliQ.model.NetworkInterface import NetworkInterface
 
 __log__ = logging.getLogger(__name__)
 
 env_id = 180005401 # CHANGE THIS TO LOCAL iQ ENVIRONMENT ID
-vm_uuid = "421be947-4e42-3deb-e0e9-55ef78c3e16a" # CHANGE THIS TO LOCAL VM UUID
+vm_hwid = "00:50:56:9B:7C:76" # CHANGE THIS TO A LOCAL VM MAC ADDRESS, MUST USE ':' NOT '-' FOR NOW
```
Comment (thorste) [inline scripts/report_event.py]: The latest Signal-iQ library should handle the dashes case now.

File: scripts/Send-Signal.ps1 [hunk 2]
```diff
@@ -1,67 +1,136 @@
...
+        }
+    }
+}
 
-$pyscript = ".\report_event.py"
-$node = $env:COMPUTERNAME
+### ENTRY POINT #######################################################################################################
 
-$xMinutesAgo = (Get-Date).Subtract((New-TimeSpan -Minutes $MinutesPrevious))
+# parse the current epoch time as a signed long, then create DateTime object from it
```
Comment (thorste) [inline scripts/Send-Signal.ps1:69]: Nice hack :)

File: scripts/Send-Signal.ps1 [hunk 3]
```diff
@@ -1,67 +1,131 @@
...
+                1 { $evtSeverity = "Info" }
+                2 { $evtSeverity = "Warning" }
+                3 { $evtSeverity = "Critical" }
+                default { $evtSeverity = "Info" }
+            }
+            
+            $tz = Get-TimeZone
+            $tzinfo = $tz.BaseUtcOffset.Hours.ToString("00") + [Math]::abs($tz.BaseUtcOffset.Minutes).ToString("00")
+
+            Write-Verbose ("Sending the following event details to the python script:`n" + $Source + "`n" + $evt.EventID + "`n" + $evtSeverity + "`n" + $evt.Message + "`n" + ($evt.TimeGenerated.ToString("yyyy-MM-ddTHH:mm:ss") + $tzinfo))
```
Comment (travisgoldie) [inline scripts/Send-Signal.ps1:41]: So the parenthesis are optional?

**Comment by srazin** [comment]
Is there a plan to submit the screenshots?

**Comment by carrollh** [comment]
Once I finish testing these changes work with the latest appliance updates. I tend to do screenshot stuff last so I don't waste time on them.

**Comment by srazin** [comment]
One more note. IMPORTANT: There is one more property that we will need to send and provide to the user which is "source". Source will be a searchable parameter. Which means that if the source is datakeeper I can find them all, etc.

Please add that parameter and @travisgoldie will make a chance to accept that.

**Comment by carrollh** [comment]
Ok. The way I've been handling that up to now is by making the source the first part of the message field. Which means it's already searchable with enough sql foo. I've been using 
select * from ... where description like 'ExtMirr%' order by time desc;
Having it searchable as its own column sounds good though. Can I add that to the next pull request? I'd like to get this one closed before working on the changes we discussed yesterday. It's long enough already IMHO.

File: scripts/report_event.py (hunk 1 above)
Comment (carrollh) [reply scripts/report_event.py]: Updated to use dashes.

**Comment by carrollh** [comment]
Can we close this now, so I can work on the next sprint items?

**Comment by srazin** [comment]
Now can you plz stuff it into the source property so it will be searchable
in iQ.

Thanks

**Comment by srazin** [comment]
Did you add the source info into the event? Travis has the support for it now. Could you please add before it call it done?

**Comment by carrollh** [comment]
I'll do that whenever Travis' changes make it into the sdk.

**Comment by srazin** [comment]
Sorry, its in now ;)

-----
Regards,
Sergey A. Razin, Ph.D. |  Chief Technology Officer
SIOS Technology Corp.
P: 650.669.8914
E: sergey.razin@us.sios.com
LinkedIn: http://www.linkedin.com/in/sergeyrazin
Twitter @ TechDozor <https://twitter.com/techdozor>
www.us.sios.com

**Comment by carrollh** [comment]
Yeah, I walked over to ask him about it. He apparently pushed it in between me sending the last message and me asking him...

**Comment by srazin** [comment]
concurrency issue ;)

-----
Regards,
Sergey A. Razin, Ph.D. |  Chief Technology Officer
SIOS Technology Corp.
P: 650.669.8914
E: sergey.razin@us.sios.com
LinkedIn: http://www.linkedin.com/in/sergeyrazin
Twitter @ TechDozor <https://twitter.com/techdozor>
www.us.sios.com

File: scripts/Send-Signal.ps1 (hunk 3 above)
Comment (carrollh) [reply scripts/Send-Signal.ps1:41]: For the most part not really. Can you be more specific?

Comment (travisgoldie) [reply scripts/Send-Signal.ps1:41]: Here you wrap the string in parenthesis. Farther down at the other `Write-Verbose` call, there are no parenthesis. So thus the question. More a question on syntax.

Comment (carrollh) [reply scripts/Send-Signal.ps1:41]: Ok. I couldn't tell what you were talking about without knowing that you were referencing another call to Write-Verbose further down in the file.
The Write-Verbose cmdlet takes a single string as input to the -Message flag. Normally PS cmdlets _don't_ use parens, they use flags followed by a single parameter input. In this case I needed to combine a lot of little strings into one, so I needed the parens here to make a single parameter. The flag part (in this case -Message) is usually optional and/or order dependent if more than one param is required.

**Review by srazin — APPROVED** [review]
Looks good.

**Pull request merged**